src/picongpu/include/fields/FieldJ.hpp

```diff
@@ -152,17 +152,19 @@ class FieldJ : public SimulationFieldHelper<MappingDesc>, public ISimulationData
 template<typename T_SpeciesName, typename T_Area>
 struct ComputeCurrent
 {
+    using SpeciesName = T_SpeciesName;
+    using SpeciesType = typename SpeciesName::type;
 
-    template<typename T_StorageTuple>
-    HINLINE void operator()( FieldJ* fieldJ,
-                            T_StorageTuple& tuple,
-                            const uint32_t currentStep) const
+    HINLINE void operator()( const uint32_t currentStep ) const
     {
-        typedef T_SpeciesName SpeciesName;
-        typedef typename SpeciesName::type SpeciesType;
+        DataConnector &dc = Environment<>::get().DataConnector();
+        auto species = dc.get< SpeciesType >( SpeciesType::FrameType::getName(), true );
```

Comment:
species name is already avail: please use `dc.get< SpeciesType >( SpeciesName(), true );`